mergin/client.py

```diff
@@ -761,8 +761,11 @@ def set_project_access(self, project_path, access):
         """
         Updates access for given project.
         :param project_path: project full name (<namespace>/<name>)
-        :param access: dict <readersnames, writersnames, ownersnames> -> list of str username we want to give access to
+        :param access: dict <readersnames, editorsnames, writersnames, ownersnames> -> list of str username we want to give access to
         """
+        if not is_version_acceptable(self.server_version(), "2024.4"):
+            raise NotImplementedError("This needs server at version 2024.4.0 or later")
```

Comment:
I would suggest to raise this only when the `access` dictionary contains `editorsnames` so that we do not break things for older servers unnecessarily.

Also, I would suggest to add to docstring that `editorsnames` is only supported by servers >= 2024.4